From b03a49dd619e4496a2f4db67c2d7c8c44fc80567 Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Sat, 3 Aug 2019 11:56:29 +0800
Subject: [PATCH] Add iptable relevant rules

Add iptable relevant rules, since WRLinux1019 product use iptable instead of firewalld.

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../iptables_input_reject_rule/bash/wrlinux1019.sh |   4 +
 .../oval/wrlinux1019.xml                           |  35 +++++
 .../iptables_input_reject_rule/rule.yml            |  21 +++
 .../oval/wrlinux1019.xml                           | 141 +++++++++++++++++++++
 .../package_iptables_installed/rule.yml            |  18 +++
 .../profiles/draft_stig_wrlinux_disa.profile       |   3 +-
 6 files changed, 221 insertions(+), 1 deletion(-)
 create mode 100644 linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/iptables_input_reject_rule/bash/wrlinux1019.sh
 create mode 100644 linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/iptables_input_reject_rule/oval/wrlinux1019.xml
 create mode 100644 linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/iptables_input_reject_rule/rule.yml
 create mode 100644 linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/package_iptables_installed/oval/wrlinux1019.xml
 create mode 100644 linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/package_iptables_installed/rule.yml

diff --git a/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/iptables_input_reject_rule/bash/wrlinux1019.sh b/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/iptables_input_reject_rule/bash/wrlinux1019.sh
new file mode 100644
index 0000000..f09bc9f
--- /dev/null
+++ b/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/iptables_input_reject_rule/bash/wrlinux1019.sh
@@ -0,0 +1,4 @@
+# platform = multi_platform_wrlinux
+
+/usr/sbin/iptables -A INPUT -j REJECT --reject-with icmp-host-prohibited
+/usr/sbin/iptables-save > /etc/sysconfig/iptables
diff --git a/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/iptables_input_reject_rule/oval/wrlinux1019.xml b/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/iptables_input_reject_rule/oval/wrlinux1019.xml
new file mode 100644
index 0000000..579a8ae
--- /dev/null
+++ b/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/iptables_input_reject_rule/oval/wrlinux1019.xml
@@ -0,0 +1,35 @@
+<def-group>
+  <definition class="compliance" id="iptables_input_reject_rule" version="1">
+    <metadata>
+      <title>Add Reject Rule for
+      the INPUT built-in chain</title>
+      <affected family="unix">
+		<platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>Add a reject rule 
+      for the INPUT built-in chain.</description>
+    </metadata>
+
+    <criteria comment="package iptables installed and the reject rule for the INPUT built-in chain is added" operator="AND">
+      <extend_definition comment="iptables installed" definition_ref="package_iptables_installed" />
+      <criteria comment="The reject rule for the INPUT built-in chain is added">
+        <criterion comment="-A INPUT -j REJECT --reject-with icmp-host-prohibited"
+        test_ref="test_iptables_input_reject_rule" />
+      </criteria>
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test check="all"
+  check_existence="at_least_one_exists"
+  comment="Check /etc/sysconfig/iptables for line -A INPUT -j REJECT --reject-with icmp-host-prohibited"
+  id="test_iptables_input_reject_rule" version="1">
+    <ind:object object_ref="obj_iptables_input_reject_rule" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="obj_iptables_input_reject_rule"
+  version="1">
+    <ind:path>/etc/sysconfig</ind:path>
+    <ind:filename>iptables</ind:filename>
+    <ind:pattern operation="pattern match">^(-[I|A]\s+(?:[A-Za-z0-9\-_]*)*INPUT\s+-j\s+REJECT\s+--reject-with\s+icmp-host-prohibited\s*)$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+</def-group>
diff --git a/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/iptables_input_reject_rule/rule.yml b/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/iptables_input_reject_rule/rule.yml
new file mode 100644
index 0000000..c043da9
--- /dev/null
+++ b/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/iptables_input_reject_rule/rule.yml
@@ -0,0 +1,21 @@
+documentation_complete: true
+
+title: 'Add a reject rule for the INPUT built-in chain'
+
+description: |-
+    To add a reject rule for the built-in INPUT chain which processes incoming packets,
+    add or correct the following line in
+    <tt>/etc/sysconfig/iptables</tt>:
+    <pre>:INPUT -j REJECT --reject-with icmp-host-prohibited</pre>
+
+rationale: |-
+    In <tt>iptables</tt> add a reject rule for the INPUT chain.
+
+severity: medium
+
+ocil_clause: 'the default policy for the INPUT chain is not set to DROP'
+
+ocil: |-
+    Inspect the file <tt>/etc/sysconfig/iptables</tt> to determine
+    the reject rule for the INPUT chain:
+    <pre>$ sudo grep ":INPUT" /etc/sysconfig/iptables</pre>
diff --git a/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/package_iptables_installed/oval/wrlinux1019.xml b/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/package_iptables_installed/oval/wrlinux1019.xml
new file mode 100644
index 0000000..56c83b0
--- /dev/null
+++ b/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/package_iptables_installed/oval/wrlinux1019.xml
@@ -0,0 +1,141 @@
+<def-group>
+  <definition class="compliance" id="package_iptables_installed"
+  version="1">
+    <metadata>
+      <title>Package iptables Installed</title>
+      <affected family="unix">
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The iptables package is installed</description>
+    </metadata>
+    <criteria operator="AND">
+      <criterion comment="The /usr/sbin/xtables-legacy-multi file is present" test_ref="test_usr_sbin_xtables_legacy_multi_is_present" />
+      <criterion comment="The /usr/sbin/iptables file is present" test_ref="test_usr_sbin_iptables_is_present" />
+      <criterion comment="The /usr/sbin/iptables-save file is present" test_ref="test_usr_sbin_iptables_save_is_present" />
+      <criterion comment="The /usr/sbin/iptables-restore file is present" test_ref="test_usr_sbin_iptables_restore_is_present" />
+      <criterion comment="The /usr/sbin/ip6tables file is present" test_ref="test_usr_sbin_ip6tables_is_present" />
+      <criterion comment="The /usr/sbin/ip6tables-save file is present" test_ref="test_usr_sbin_ip6tables_save_is_present" />
+      <criterion comment="The /usr/sbin/ip6tables-restore file is present" test_ref="test_usr_sbin_ip6tables_restore_is_present" />
+    </criteria>
+  </definition>
+
+
+  <unix:file_test id="test_usr_sbin_xtables_legacy_multi_is_present"
+	check="all"
+	check_existence="all_exist"
+	comment="The /usr/sbin/xtables-legacy-multi file is present on the system"
+	version="1"
+	>
+    <unix:object object_ref="object_usr_sbin_xtables_legacy_multi_is_present" />
+  </unix:file_test>
+
+  <unix:file_object id="object_usr_sbin_xtables_legacy_multi_is_present"
+	comment="The /usr/sbin/xtables-legacy-multi file is present on the system"
+	version="1"
+	>
+    <unix:filepath operation="equals">/usr/sbin/xtables-legacy-multi</unix:filepath>
+  </unix:file_object>
+
+
+  <unix:file_test id="test_usr_sbin_iptables_is_present"
+	check="all"
+	check_existence="all_exist"
+	comment="The /usr/sbin/iptables file is present on the system"
+	version="1"
+	>
+    <unix:object object_ref="object_usr_sbin_iptables_is_present" />
+  </unix:file_test>
+
+  <unix:file_object id="object_usr_sbin_iptables_is_present"
+	comment="The /usr/sbin/iptables file is present on the system"
+	version="1"
+	>
+    <unix:filepath operation="equals">/usr/sbin/iptables</unix:filepath>
+  </unix:file_object>
+
+
+  <unix:file_test id="test_usr_sbin_iptables_save_is_present"
+	check="all"
+	check_existence="all_exist"
+	comment="The /usr/sbin/iptables-save file is present on the system"
+	version="1"
+	>
+    <unix:object object_ref="object_usr_sbin_iptables_save_is_present" />
+  </unix:file_test>
+
+  <unix:file_object id="object_usr_sbin_iptables_save_is_present"
+	comment="The /usr/sbin/iptables-save file is present on the system"
+	version="1"
+	>
+    <unix:filepath operation="equals">/usr/sbin/iptables-save</unix:filepath>
+  </unix:file_object>
+
+
+  <unix:file_test id="test_usr_sbin_iptables_restore_is_present"
+	check="all"
+	check_existence="all_exist"
+	comment="The /usr/sbin/iptables-restore file is present on the system"
+	version="1"
+	>
+    <unix:object object_ref="object_usr_sbin_iptables_restore_is_present" />
+  </unix:file_test>
+
+  <unix:file_object id="object_usr_sbin_iptables_restore_is_present"
+	comment="The /usr/sbin/iptables-restore file is present on the system"
+	version="1"
+	>
+    <unix:filepath operation="equals">/usr/sbin/iptables-restore</unix:filepath>
+  </unix:file_object>
+
+
+  <unix:file_test id="test_usr_sbin_ip6tables_is_present"
+	check="all"
+	check_existence="all_exist"
+	comment="The /usr/sbin/ip6tables file is present on the system"
+	version="1"
+	>
+    <unix:object object_ref="object_usr_sbin_ip6tables_is_present" />
+  </unix:file_test>
+
+  <unix:file_object id="object_usr_sbin_ip6tables_is_present"
+	comment="The /usr/sbin/ip6tables file is present on the system"
+	version="1"
+	>
+    <unix:filepath operation="equals">/usr/sbin/ip6tables</unix:filepath>
+  </unix:file_object>
+
+
+  <unix:file_test id="test_usr_sbin_ip6tables_save_is_present"
+	check="all"
+	check_existence="all_exist"
+	comment="The /usr/sbin/ip6tables-save file is present on the system"
+	version="1"
+	>
+    <unix:object object_ref="object_usr_sbin_ip6tables_save_is_present" />
+  </unix:file_test>
+
+  <unix:file_object id="object_usr_sbin_ip6tables_save_is_present"
+	comment="The /usr/sbin/ip6tables-save file is present on the system"
+	version="1"
+	>
+    <unix:filepath operation="equals">/usr/sbin/ip6tables-save</unix:filepath>
+  </unix:file_object>
+
+
+  <unix:file_test id="test_usr_sbin_ip6tables_restore_is_present"
+	check="all"
+	check_existence="all_exist"
+	comment="The /usr/sbin/ip6tables-restore file is present on the system"
+	version="1"
+	>
+    <unix:object object_ref="object_usr_sbin_ip6tables_restore_is_present" />
+  </unix:file_test>
+
+  <unix:file_object id="object_usr_sbin_ip6tables_restore_is_present"
+	comment="The /usr/sbin/ip6tables-restore file is present on the system"
+	version="1"
+	>
+    <unix:filepath operation="equals">/usr/sbin/ip6tables-restore</unix:filepath>
+  </unix:file_object>
+
+</def-group>
diff --git a/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/package_iptables_installed/rule.yml b/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/package_iptables_installed/rule.yml
new file mode 100644
index 0000000..f21ef2c
--- /dev/null
+++ b/linux_os/guide/system/network/network-iptables/iptables_ruleset_modifications/package_iptables_installed/rule.yml
@@ -0,0 +1,18 @@
+documentation_complete: true
+
+prodtype: wrlinux1019
+
+title: 'Install the iptables Package'
+
+description: |-
+    The <tt>iptables</tt> package should be installed.
+    {{{ describe_package_install(package="iptables") }}}
+
+rationale: |-
+    The iptables package must be installed if it is to be available for integrity checking.
+
+severity: medium
+
+ocil_clause: 'the package iptables is not installed'
+
+ocil: '{{{ ocil_package(package="iptables") }}}'
diff --git a/wrlinux1019/profiles/draft_stig_wrlinux_disa.profile b/wrlinux1019/profiles/draft_stig_wrlinux_disa.profile
index 7495a83..6e5a73d 100644
--- a/wrlinux1019/profiles/draft_stig_wrlinux_disa.profile
+++ b/wrlinux1019/profiles/draft_stig_wrlinux_disa.profile
@@ -214,7 +214,8 @@ selections:
     - selinux_all_devicefiles_labeled
     - accounts_umask_interactive_users
     - rsyslog_nolisten
-    - service_firewalld_enabled
+    - package_iptables_installed
+    - iptables_input_reject_rule
     - package_tftp-server_removed
     - configure_firewalld_ports
     - sssd_ldap_start_tls
-- 
1.9.1

